web-registry/src/components/organisms/BasketsTable.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Box, Grid } from '@mui/material';
+
+import {
+  ActionsTable,
+  renderActionButtonsFunc,
+} from 'web-components/lib/components/table/ActionsTable';
+
+import { NoEcocredits } from '../molecules';
+import type { TableBaskets } from '../../types/ledger/ecocredit';
+import { ReactComponent as BasketIcon } from '../../assets/svgs/rNCT.svg';
+
+export const BasketsTable: React.FC<{
+  baskets: TableBaskets[];
+  renderActionButtons?: renderActionButtonsFunc;
+}> = ({ baskets, renderActionButtons }) => {
+  if (!baskets?.length) {
+    return <NoEcocredits title="No basket tokens to display" />;
+  }
+  return (
+    <ActionsTable
+      tableLabel="baskets table"
+      renderActionButtons={renderActionButtons}
+      headerRows={[
+        <Box
+          sx={{
+            minWidth: {
+              xs: 'auto',
+              sm: '11rem',
+              lg: '13rem',
+            },
+          }}
+        >
+          Asset
+        </Box>,
+        'Balance',
+      ]}
+      rows={baskets.map((row, i) => {
+        return [
+          <Grid container wrap="nowrap">
+            <Grid item>
+              <BasketIcon />
+            </Grid>
+            <Grid item sx={{ ml: 3.5, fontWeight: 700 }} alignSelf="center">
+              <Box sx={{ fontSize: theme => theme.spacing(4.5) }}>
+                {row.name}
+              </Box>
+              <Box sx={{ color: 'info.main' }}>
+                {row.display_denom.toLocaleString()}
+              </Box>
+            </Grid>
+          </Grid>,
+          row.balance?.amount
+            ? parseInt(row.balance?.amount) /
+              Math.pow(10, parseInt(row.exponent))
+            : 0,
```

Comment:
are these numbers the same? if so and `exponent` is always available, maybe we should just use that?

Reply:
a basket token always has an exponent defined, let's called it `exp`, meaning that for `x` ecocredits you put into this basket, you then get `x * 10^exp` basket tokens back. 
these basket tokens are tracked as regular coins (just like `regen`s) in the bank module as integers so when you query someone's balance for this basket token, you'll get the amount `x * 10^exp`. For the rNCT token, the corresponding denom will be `eco.uC.rNCT` (notice the `u` which corresponds to `micro`, meaning the exponent is 6, just like in `uregen`). But on the UI we actually want to display the amount in `eco.C.rNCT` which corresponds to the initial amount of ecocredits so we need to divide the amount we get in `eco.uC.rNCT` from the bank module by `10^exp`

See the docs on regen-ledger: https://github.com/regen-network/regen-ledger/blob/release/v3.0.x/proto/regen/ecocredit/basket/v1/tx.proto#L44-L68

Reply:
That makes sense - thanks for the explanation!